lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/restart/DFSEvaluatorLogAppendReaderWriter.java

```diff
@@ -65,6 +66,11 @@ public synchronized void writeToEvaluatorLog(final String formattedEntry) throws
     }
   }
 
+  @Override
+  public Iterable<String> readFromEvaluatorLog() throws IOException {
```

Comment:
Right. But this design assumes that the whole file is read, which can leak resources.

I like the `ClosableIterable` as it would lead to code like this:

```java
try(ClosableIterable<String> lines = someObject.openFile()){
  for(final string line: lines){...}
}
catch(IOException e){...}
catch(ParseException e){...}
```

This is pretty clean and makes it obvious that the file needs to be closed at the end.